src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ConfigurationList.cs

```diff
@@ -3,30 +3,33 @@
 
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics;
 
 namespace System.Text.Json.Serialization
 {
     /// <summary>
-    /// A list of JsonConverters that respects the options class being immuttable once (de)serialization occurs.
+    /// A list of configuration items that respects the options class being immutable once (de)serialization occurs.
     /// </summary>
-    internal sealed class ConverterList : IList<JsonConverter>
+    internal sealed class ConfigurationList<TItem> : IList<TItem>
```

Comment:
I fail to see why this cannot use `List<T>` and once you're done mutating calling `AsReadOnly`

Reply:
`AsReadOnly()` returns a read-only wrapper of the list and doesn't lock the original instance.